File: application/app/src/main/java/com/yahyeet/boardbook/presenter/matchcreation/selectgame/SelectGamePresenter.java
package com.yahyeet.boardbook.presenter.matchcreation.selectgame;

import android.content.Context;

import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.yahyeet.boardbook.activity.IFutureInteractable;
import com.yahyeet.boardbook.model.entity.Game;
import com.yahyeet.boardbook.model.handler.GameHandler;
import com.yahyeet.boardbook.presenter.FindAllPresenter;
import com.yahyeet.boardbook.presenter.BoardbookSingleton;
import com.yahyeet.boardbook.activity.matchcreation.selectgame.ISelectGameFragment;
import com.yahyeet.boardbook.presenter.matchcreation.CMMasterPresenter;

import java.util.List;

/**
 * This is the Presenter for the SelectGame fragment.
 * This class binds references to the MasterPresenter, Enables the RecycleView and gives it the correct adapter
 */
public class SelectGamePresenter extends FindAllPresenter<Game, GameHandler> {

	private ISelectGameFragment selectGameFragment;
	private CMMasterPresenter masterPresenter;


	public SelectGamePresenter(ISelectGameFragment selectGameFragment, CMMasterPresenter cm) {
		super((IFutureInteractable) selectGameFragment);
		this.selectGameFragment = selectGameFragment;
		this.masterPresenter = cm;

		setAdapter(new GamesAdapter(getDatabase(), this));
		fillAndModifyDatabase(BoardbookSingleton.getInstance().getGameHandler(),
			GameHandler.generatePopulatorConfig(false, true));


	}

	public void enableGameList(RecyclerView gameRecycleView, Context viewContext) {
		RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(viewContext);
		gameRecycleView.setLayoutManager(layoutManager);
		gameRecycleView.setAdapter(getAdapter());
	}

	public CMMasterPresenter getMasterPresenter(){
		return masterPresenter;
	}


	@Override
	protected void onDatabaseModify(List<Game> database) {
		selectGameFragment.enableRecyclerList();
	}
}

File: application/app/src/main/java/com/yahyeet/boardbook/presenter/matchcreation/selectgame/SelectGamePresenter.kt
package com.yahyeet.boardbook.presenter.matchcreation.selectgame

import android.content.Context

import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView

import com.yahyeet.boardbook.model.entity.Game
import com.yahyeet.boardbook.presenter.BoardbookSingleton
import com.yahyeet.boardbook.presenter.matchcreation.CMMasterPresenter
import com.yahyeet.boardbook.activity.matchcreation.selectgame.ISelectGameFragment

import java.util.ArrayList

class SelectGamePresenter(private val selectGameFragment: ISelectGameFragment, val masterPresenter: CMMasterPresenter) {

    private var gamesAdapter: GamesAdapter? = null
    private val dataset = ArrayList<Game>()

    fun updateAdapter() {
        gamesAdapter!!.notifyDataSetChanged()
    }

    fun enableGameFeed(gameRecycleView: RecyclerView, viewContext: Context) {
        val layoutManager = LinearLayoutManager(viewContext)
        gameRecycleView.layoutManager = layoutManager
        /*Game testGame = new Game("Avalon", "Cool Game", 3, 5, 10);
		GameTeam mom = new GameTeam();
		mom.setName("Minions of Mordred");
		GameRole mordred = new GameRole();
		mordred.setName("Mordred");
		GameRole minion = new GameRole();
		minion.setName("Minion of Mordred");
		mom.addRole(mordred);
		mom.addRole(minion);
		testGame.addTeam(mom);
		GameTeam som = new GameTeam();
		som.setName("Servants of Merlin");
		GameRole merlin = new GameRole();
		merlin.setName("Merlin");
		GameRole servant = new GameRole();
		servant.setName("Servant of Merlin");
		som.addRole(merlin);
		som.addRole(servant);
		testGame.addTeam(som);


		List<Game> testSet = new ArrayList<>();
		testSet.add(testGame);*/



        gamesAdapter = GamesAdapter(dataset, this)
        gameRecycleView.adapter = gamesAdapter
    }

}
